js/rendererjs/pluginapi.js

```diff
@@ -2,13 +2,52 @@
 // This is injected into every plugin's global namespace.
 import Siad from 'sia.js'
 import { remote } from 'electron'
+import React from 'react'
+import DisabledPlugin from './disabledplugin.js'
 const dialog = remote.dialog
 const mainWindow = remote.getCurrentWindow()
 const config = remote.getGlobal('config')
 Siad.configure(config.siad)
+let disabled = false
+
+window.onload = () => {
+	// ReactDOM needs a DOM in order to be imported,
+	// but the DOM is not available until the plugin has loaded.
+	// therefore, we have to global require it inside the window.onload event.
+
+	/* eslint-disable global-require */
+	const ReactDOM = require('react-dom')
+	/* eslint-enable global-require */
+
+	const checkSiad = () => {
+		Siad.ifRunning(() => {
+			if (disabled) {
+				disabled = false
+				window.location.reload()
```

Comment:
This polls siad continuously (every 2s), and disables the plugin if siad is not running.  The plugin is 'disabled' by mounting <DisabledPlugin /> in the DOM's body.  If siad is found to be running, and the plugin is disabled, the window is reloaded, clearing the <DisabledPlugin /> component and reloading the plugin.